src/tests/index.test.js

```diff
@@ -0,0 +1,549 @@
+                const expArr = ["Javascript", "is", "awesome"];
+                const expSize = 3;
+
+                ArrayList.init(arr);
+
+                const actual = ArrayList.init(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize);
+            });
+        });
+    });
+    describe('Test AdAtEnd method', () => {
+        describe('ArrayList.init', function () {
+            it('should be defined', function () {
+                assert.isDefined(ArrayList.addAtEnd, 'method has been defined')
+            });
+
+            it('should warning user to [] (null)', () => {
+                const arr = null;
+                const expArr = 'Please input an array';
+                const expSize = 'Please input an array';
+
+                const actual = ArrayList.addAtEnd(arr);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual, expSize)
+            });
+
+            it('should warning user to [] (undefined)', () => {
+                const arr = undefined;
+                const expArr = 'Please input an array';
+                const expSize = 'Please input an array';
+
+                const actual = ArrayList.addAtEnd(arr);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual, expSize)
+            });
+
+            it('should return an array [1,2,3,4]', () => {
+                const arr = [1,2,3];
+                const arrElm = 4;
+                const expArr = [1,2,3,4];
+                const expSize = 4;
+                ArrayList.init(arr);
+
+                const actual = ArrayList.addAtEnd(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize)
+                
+            });
+
+            it('should return "Javascript is awesome', () => {
+                const arr = ['Javascript', 'is'];
+                const arrElm = 'awesome';
+                const expArr = ['Javascript', 'is', 'awesome'];
+                const expSize = 3;
+                ArrayList.init(arr);
+
+                const actual = ArrayList.addAtEnd(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize)
+            });
+        });
+    });
+    describe('Test clearObject', () => {
+        describe('ArrayList.clearObject', () => {
+            it('should be defined', () => {
+                assert.isDefined(ArrayList.clearObject, 'method has been defined')
+            });
+
+            it('should return []', () => {
+                const arr = [1,2,3];
+                const expArr = [];
+                const expSize = 0;
+                ArrayList.init(arr);
+
+                const actual = ArrayList.clearObject();
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize)
+            });
+        });
+    });
```

Comment:
you could check more test cases than 2. We agreed that you should check the empty array, array with one element, array with two elements. It's common rules, please try stick to the rules )